src/Development/IDE/Spans/Type.hs

```diff
@@ -34,9 +35,11 @@ data SpanInfo =
             -- any. This can be useful for accessing a variety of
             -- information about the identifier such as module,
             -- locality, definition location, etc.
+           ,spaninfoDocs :: ![T.Text]
```

Comment:
`SpanInfo` is cached so everything that we store here is retained in memory. The way this is setup currently, you convert the docs for a symbol separately for each module. This means that if module `A` defines symbol `a` and module `B` and `C` reference it, you will end up storing the docs for `a` twice. For such a small example, this is probably fine but on larger projects this seems like something that could cause problems.

I would suggest that we only cache the `HsDocString` here which should be shared and then do the conversion to `Text` (and markdown) on the fly (probably in `atPoint`).

Reply:
Good idea! I'll do this in this PR, and then do something similar for completions, where the string may be also shared.